prog-o-meter.py

```diff
@@ -146,7 +152,39 @@ def add_day(self):
         update_days_file(self.filename, self.days)
         self.canvas.itemconfig(self.countdown_text, text = "".join(("You have ", str(self.days_remaining), " days left!\n\n", "If you code everyday, you will be done with this project on ", self.completion_date)))
         if self.days >=self.GOAL:        # Disable add_day_button if goal have been reached 
-            self.add_day_button.config(state = "disabled")             
+            self.add_day_button.config(state = "disabled")  
+    def log_entry(self):
+        """Opens a new window for user to make a new log entry. The user can make any number of entries they wish.
+
+        Callback function to add_log_button. Opens a new toplevel window with a text widget and three buttons:
+        A button marked "Save" that saves the text the user typed into the widget to the [USERNAME]_log.txt file.
+        A button marked "Clear" that clears the text that the user typed into the widget.
+        A button marked "Close" that closes the toplevel window.
+        """
+        log_window = Tk.Toplevel(self.root)
+
+        log_window.title("Write your log entry here:")
+
+        scroll_bar = Tk.Scrollbar(log_window)         #A vertical scroll bar for user's convenience
+        text_box = Tk.Text(log_window, height=10, width=50)
+
+        scroll_bar.pack(side=Tk.RIGHT, fill=Tk.Y)
+        text_box.pack(fill=Tk.Y)
+
+        scroll_bar.config(command=text_box.yview)
+        text_box.config(yscrollcommand=scroll_bar.set)
+
+        def update_log():

```

Comment:
This function needs a docstring documentation (like you have for the function above on lines 157-163). 
I am also wondering if this function needs to be nested? I haven't worked much with nested functions myself, so I am not aware what is gained by having this nested?

Reply:
I've added the docstring.

It may be possible to do this without nesting the function. But this way I can use "command = lambda:" on line 185, which calls update_log() only after the user presses the "save" button. I'm not sure how it would be done if I didn't give the function update_log() its own name. 

Note that on line 182, update_log() refers to text_box. text_box is a variable defined only inside the function log_entry (on line 169). If one didn't nest update_log() inside log_entry, then I think you would have to make text_box a global variable, which I don't want to do. To avoid this issue of scope, I just nested the function inside.

Reply:
Thank you for the explanation, makes sense :)